lib/tests/fixtures/compiler/evaluation/type_aliases.pp

```
@@ -0,0 +1,13 @@
+type IntegerTree = Array[Variant[Integer, IntegerTree]]
+type Mix = Variant[Integer, String, MixedTree]
+type MixedTree = Array[Variant[Mix, MixedTree]]
+
+function integer_tree(IntegerTree $x) {
+  notice $x
+}
+integer_tree( [1, 2, [42, 4], [[[ 5 ]]] ] )
+
+function mixed(MixedTree $x) {
+  notice $x
+}
+mixed( [1, 2, [hello, 4], [[[ 5, deep ]]] ] )
```

You also need a test that a mismatch hidden in deep nesting is detected, e.g.:
```
mixed( [1, 2, [hello, 4], [[[ 'five', deep ]]] ] )
```
should raise an exception.